### PR TITLE
integrate new arwens and vm common

### PR DESCRIPTION
**Pull request by bogdan-rosianu**
- integrate new arwens that integrate the `elrond-go-core`

### CODE REVIEW

File: api/transaction/routes.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go/api/middleware"
 	"github.com/ElrondNetwork/elrond-go/api/shared"
 	"github.com/ElrondNetwork/elrond-go/api/wrapper"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline api/transaction/routes.go:16]: txSimData

File: go.sum [hunk 2]
```diff
@@ -11,33 +11,29 @@ github.com/AndreasBriese/bbloom v0.0.0-20180913140656-343706a395b7/go.mod h1:bOv
 github.com/AndreasBriese/bbloom v0.0.0-20190306092124-e2d15f34fcf9/go.mod h1:bOvUY6CB00SOBii9/FifXqc0awNKxLFCL/+pkDPuyl8=
 github.com/BurntSushi/toml v0.3.1 h1:WXkYYl6Yr3qBf1K79EBnL4mak0OimBfB0XUf9Vl28OQ=
 github.com/BurntSushi/toml v0.3.1/go.mod h1:xHWCNGjB5oqiDr8zfno3MHue2Ht5sIBksp03qcyfWMU=
-github.com/ElrondNetwork/arwen-wasm-vm v1.2.26 h1:L2es+JjdfaiKd2IxicVhkRCZfu8E1hoyEk/8zgxxmR4=
-github.com/ElrondNetwork/arwen-wasm-vm v1.2.26/go.mod h1:6TqZmq72m8T4SK7F2q8Bgvbz/FrzvXrGLfmMHt1Td+E=
-github.com/ElrondNetwork/arwen-wasm-vm v1.3.25 h1:MvNrEeSeDFKvZhl9vstEmFM4oiAIR1NlwTEyWLCT4xY=
-github.com/ElrondNetwork/arwen-wasm-vm v1.3.25/go.mod h1:2pHVIM+Fifssv04JXX6dqIorgVgerAIx8pElwd0WjvA=
-github.com/ElrondNetwork/arwen-wasm-vm v1.4.2-0.20210720092934-6b2c1c46a8ae h1:u7zVpMCtS0CeUPeMj26p/7wa4yEAR6LHnyzdvz6CdNY=
-github.com/ElrondNetwork/arwen-wasm-vm v1.4.2-0.20210720092934-6b2c1c46a8ae/go.mod h1:fD8SIuR6OhPp8yIfLkSD+sTako2+6GmGuTkakCPx7l8=
+github.com/ElrondNetwork/arwen-wasm-vm v1.2.28 h1:lhGICGe0yYlj4wquGl7t24qrHFlZhOFuhda/bWvaBe4=
```
Comment (miiu96) [inline go.sum:14]: go mod tidy ?

File: facade/interface.go [hunk 3]
```diff
@@ -5,14 +5,15 @@ import (
 
 	"github.com/ElrondNetwork/elrond-go-core/core"
 	"github.com/ElrondNetwork/elrond-go-core/data/api"
+	"github.com/ElrondNetwork/elrond-go-core/data/esdt"
 	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
 	"github.com/ElrondNetwork/elrond-go/debug"
 	"github.com/ElrondNetwork/elrond-go/heartbeat/data"
 	"github.com/ElrondNetwork/elrond-go/node/external"
 	"github.com/ElrondNetwork/elrond-go/process"
+	txSimulatorData "github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline facade/interface.go:14]: txSimData

File: facade/mock/txExecutionSimulatorStub.go [hunk 4]
```diff
@@ -2,20 +2,21 @@ package mock
 
 import (
 	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline facade/mock/txExecutionSimulatorStub.go:5]: txSimData

File: integrationTests/mock/transactionSimulatorStub.go [hunk 5]
```diff
@@ -2,15 +2,16 @@ package mock
 
 import (
 	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline integrationTests/mock/transactionSimulatorStub.go:5]: txSimData

File: integrationTests/testProcessorNodeWithTestWebServer.go [hunk 6]
```diff
@@ -19,6 +19,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go/process/smartContract/builtInFunctions"
 	"github.com/ElrondNetwork/elrond-go/process/transaction"
 	"github.com/ElrondNetwork/elrond-go/process/txsimulator"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline integrationTests/testProcessorNodeWithTestWebServer.go:22]: txSimData

File: process/mock/transactionSimulatorStub.go [hunk 7]
```diff
@@ -2,15 +2,16 @@ package mock
 
 import (
 	"github.com/ElrondNetwork/elrond-go-core/data/transaction"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline process/mock/transactionSimulatorStub.go:5]: txSimData

File: process/txsimulator/txSimulator.go [hunk 8]
```diff
@@ -12,6 +12,7 @@ import (
 	"github.com/ElrondNetwork/elrond-go-core/hashing"
 	"github.com/ElrondNetwork/elrond-go-core/marshal"
 	"github.com/ElrondNetwork/elrond-go/process"
+	"github.com/ElrondNetwork/elrond-go/process/txsimulator/data"
```
Comment (SebastianMarian) [inline process/txsimulator/txSimulator.go:15]: txSimData

File: go.sum (hunk 2 above)
Comment (bogdan-rosianu) [reply go.sum:14]: already done

File: api/transaction/routes.go (hunk 1 above)
Comment (bogdan-rosianu) [reply api/transaction/routes.go]: done

File: facade/interface.go (hunk 3 above)
Comment (bogdan-rosianu) [reply facade/interface.go]: done

File: facade/mock/txExecutionSimulatorStub.go (hunk 4 above)
Comment (bogdan-rosianu) [reply facade/mock/txExecutionSimulatorStub.go]: done

File: integrationTests/mock/transactionSimulatorStub.go (hunk 5 above)
Comment (bogdan-rosianu) [reply integrationTests/mock/transactionSimulatorStub.go]: done

File: integrationTests/testProcessorNodeWithTestWebServer.go (hunk 6 above)
Comment (bogdan-rosianu) [reply integrationTests/testProcessorNodeWithTestWebServer.go]: done

File: process/mock/transactionSimulatorStub.go (hunk 7 above)
Comment (bogdan-rosianu) [reply process/mock/transactionSimulatorStub.go]: done

File: process/txsimulator/txSimulator.go (hunk 8 above)
Comment (bogdan-rosianu) [reply process/txsimulator/txSimulator.go]: done

**Review by miiu96 — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**